pkg/deploy/filter/envoy_filter.go

```diff
@@ -0,0 +1,68 @@
+package filter
+
+import (
+	envoyhttp "github.com/envoyproxy/go-control-plane/envoy/config/filter/network/http_connection_manager/v2"
+	"github.com/gogo/protobuf/types"
+	"github.com/solo-io/gloo/projects/gloo/pkg/api/external/envoy/api/v2/config"
+	"github.com/solo-io/solo-kit/pkg/api/external/envoy/api/v2/core"
+	"github.com/solo-io/wasme/pkg/deploy"
+	"github.com/solo-io/wasme/pkg/util"
+)
+
+func MakeRemoteDataSource(uri, cluster string) *core.AsyncDataSource {
+	return &core.AsyncDataSource{
+		Specifier: &core.AsyncDataSource_Remote{
+			Remote: &core.RemoteDataSource{
+				HttpUri: &core.HttpUri{
+					Uri: uri,
+					HttpUpstreamType: &core.HttpUri_Cluster{
+						Cluster: cluster,
+					},
+					Timeout: &types.Duration{
+						Seconds: 5, // TODO: customize
+					},
+				},
+			},
+		},
+	}
+}
+
+func MakeLocalDatasource(path string) *core.AsyncDataSource {
+	return &core.AsyncDataSource{
+		Specifier: &core.AsyncDataSource_Local{
+			Local: &core.DataSource{
+				Specifier: &core.DataSource_Filename{
+					Filename: path,
+				},
+			},
+		},
+	}
+}
+
+func MakeWasmFilter(filter *deploy.Filter, dataSrc *core.AsyncDataSource) *envoyhttp.HttpFilter {
+	filterCfg := &config.WasmService{
+		Config: &config.PluginConfig{
+			Name:          filter.ID,
+			RootId:        filter.RootID,
+			Configuration: filter.Config,
+			VmConfig: &config.VmConfig{
+				Runtime: "envoy.wasm.runtime.v8", // default to v8
+				Code:    dataSrc,
+			},
+		},
+	}
+
+	// here we need to use the golang proto marshal
+	marshalledConf, err := util.MarshalStruct(filterCfg)
```

Comment:
Maybe we should add these to seperate packages called for instance: `goprotoutils` and `gogoprotoutils`

Reply:
i don't really want to make a breaking change to `go-utils`. i would relocate these funcs there, but not sure how to name them (without breaking)